### PR TITLE
Fix namespace replication queue cleanup logic

### PR DESCRIPTION
**Pull request by yux0**
## What changed?
Fix namespace replication queue cleanup logic

## Why?
When a cluster is disconnected, the ack level will not be updated. So the current clean up namespace replication queue based on the min ack level is not working.

### CODE REVIEW

File: service/worker/replicator/replicator.go [hunk 1]
```diff
@@ -172,3 +182,64 @@ func (r *Replicator) listenToClusterMetadataChange() {
 		},
 	)
 }
+
+func (r *Replicator) cleanupAckedMessages(
+	ctx context.Context,
+	ackMessageID int64,
+) (int64, error) {
+	ackLevelByCluster, err := r.namespaceReplicationQueue.GetAckLevels(ctx)
```
Comment (yycptt) [inline service/worker/replicator/replicator.go:190]: mark as low priority

File: service/worker/replicator/replicator.go [hunk 2]
```diff
@@ -172,3 +182,64 @@ func (r *Replicator) listenToClusterMetadataChange() {
...
+	minAckLevel := int64(math.MaxInt64)
+	for clusterName, ackLevel := range ackLevelByCluster {
+		if clusterName == r.clusterMetadata.GetCurrentClusterName() {
+			continue
+		}
+		if ackLevel > maxAckLevel {
+			maxAckLevel = ackLevel
+		}
+		if _, ok := connectedClusters[clusterName]; !ok {
+			continue
```
Comment (yycptt) [inline service/worker/replicator/replicator.go:206]: when do we delete ack level for disconnected clusters?

File: service/worker/replicator/replicator.go [hunk 3]
```diff
@@ -172,3 +182,64 @@ func (r *Replicator) listenToClusterMetadataChange() {
...
+
+func (r *Replicator) cleanupNamespaceReplicationQueue(
+	ctx context.Context,
+) error {
+	ctx = headers.SetCallerInfo(ctx, headers.SystemPreemptableCallerInfo)
+
+	ticker := time.NewTicker(replicationQueueCleanupInterval)
+	defer ticker.Stop()
+
+	ackMessageID := persistence.EmptyQueueMessageID
```
Comment (yycptt) [inline service/worker/replicator/replicator.go:232]: so this is deletedMessageID?

File: service/worker/replicator/replicator.go (hunk 2 above)
Comment (yux0) [reply service/worker/replicator/replicator.go:206]: No in this PR. Do you want to delete the ack levels? Or we just migrate to queue v2?

File: service/worker/replicator/replicator.go (hunk 3 above)
Comment (yux0) [reply service/worker/replicator/replicator.go:232]: Let me update the naming

File: service/worker/replicator/replicator.go (hunk 1 above)
Comment (yux0) [reply service/worker/replicator/replicator.go:190]: Is L227 good enough?

File: service/worker/replicator/replicator.go (hunk 2 above)
Comment (yux0) [reply service/worker/replicator/replicator.go:206]: Added a TODO, I need to expose new persistence API to support this.

File: service/worker/replicator/replicator.go [hunk 4]
```diff
@@ -172,3 +182,65 @@ func (r *Replicator) listenToClusterMetadataChange() {
...
+		if clusterName == r.clusterMetadata.GetCurrentClusterName() {
+			continue
+		}
+		if ackLevel > highWatermark {
+			highWatermark = ackLevel
+		}
+		if _, ok := connectedClusters[clusterName]; !ok {
+			continue
+		}
+		if ackLevel < connectedClustersLowWatermark {
```
Comment (yycptt) [inline service/worker/replicator/replicator.go:208]: Is ackLevel inclusive or exclusive? If inclusive, we need to + 1 when deleting messages?

Comment (yux0) [reply service/worker/replicator/replicator.go:208]: historical reason this is exclusive

**Review by yycptt — APPROVED** [review]

**Pull request merged**